### PR TITLE
Add Decision Issue and End Product Establishment FK to Request Issues table

### PR DESCRIPTION
**Pull request by wdrougas-tista**
### Description
Add Decision Issue and End Product Establishment foreign key to request issues table

* This is phase 7b of a [multi-phase plan](https://hackmd.io/wum2LtUfSHC-1o6-peBseQ) to add missing foreign keys.
* Phase 7 PR #16277 
* Phase 5c PR #16263.
* Phase 5b PR #16281.
* Phase 4 PR #16137.
* Phase 3 PR #16120.
* Phase 2 PR #16114.
* Phase 1 PR #16034 has more details.

### Database Changes
*Only for Schema Changes*

* [x] Have your migration classes inherit from `Caseflow::Migration`, especially when adding indexes (use `add_safe_index`)
* [x] Verify that `migrate:rollback` works as desired ([`change` supported functions](https://edgeguides.rubyonrails.org/active_record_migrations.html#using-the-change-method))
* [x] #appeals-schema notified with summary and link to this PR

### Testing Plan
Before deployment, rerun the following in Prod, UAT, and Preprod environments to ensure the foreign keys will be valid:

```ruby
RequestIssue.unscoped.includes(:contested_decision_issue).where.not(contested_decision_issue_id: nil).select{|r| r.contested_decision_issue == nil}.pluck(:id)
=> []
RequestIssue.unscoped.includes(:end_product_establishment).where.not(end_product_establishment_id: nil).select{|r| r.end_product_establishment == nil}.pluck(:id)
=> []
```

### CODE REVIEW

**Comment by yoomlam** [comment]
When checking for orphaned records (in all the environments), we should query using `unscoped`. [Slack](https://dsva.slack.com/archives/C013GETU45V/p1622576716060100?thread_ts=1622574187.057600&cid=C013GETU45V)

**Comment by yoomlam** [comment]
Do we have a follow-up to these action item:

```ruby
RequestIssue.includes(:contested_decision_issue).where.not(contested_decision_issue_id: nil).select{|r| r.contested_decision_issue == nil}.pluck(:id)
=> [5183, 972639, 1758655, 972638, 1932358, 1591821, 1758654]
# 2 more have popped up
# ACTION: identify root cause; if it's in our code, we need to fix it before deploying FK
```

```ruby
RequestIssue.includes(:end_product_establishment).where.not(end_product_establishment_id: nil).select{|r| r.end_product_establishment == nil}.pluck(:id)
# Wednesday, April 28
[766681, 767166, 767171, 767172, 767534, 767540, 767541, 768069, 780817, 780820, 780949, 780950, 780951, 780955, 780956, 780957, 780958, 780959, 780960, 780961, 780963, 780964, 780965, 780966, 780967, 780968, 780969, 780974, 780975, 782647, 784263, 784266, 784336, 784339, 784907, 784993, 785162, 785239, 767542, 780952, 780962, 780954]
# Will: almost positive these can be deleted; hasn't happened in 16 months
# We can create a separate PR for this one FK, in case we need to revert
# Ask Sally: If we added a FK, what would break?
# ACTION: why is this happening? 
```

**Comment by wdrougas-tista** [comment]
> Do we have a follow-up or answer to these action item:
> 
> ```ruby
> RequestIssue.includes(:contested_decision_issue).where.not(contested_decision_issue_id: nil).select{|r| r.contested_decision_issue == nil}.pluck(:id)
> => [5183, 972639, 1758655, 972638, 1932358, 1591821, 1758654]
> # 2 more have popped up
> # ACTION: identify root cause; if it's in our code, we need to fix it before deploying FK
> ```
> 
> ```ruby
> RequestIssue.includes(:end_product_establishment).where.not(end_product_establishment_id: nil).select{|r| r.end_product_establishment == nil}.pluck(:id)
> # Wednesday, April 28
> [766681, 767166, 767171, 767172, 767534, 767540, 767541, 768069, 780817, 780820, 780949, 780950, 780951, 780955, 780956, 780957, 780958, 780959, 780960, 780961, 780963, 780964, 780965, 780966, 780967, 780968, 780969, 780974, 780975, 782647, 784263, 784266, 784336, 784339, 784907, 784993, 785162, 785239, 767542, 780952, 780962, 780954]
> # Will: almost positive these can be deleted; hasn't happened in 16 months
> # We can create a separate PR for this one FK, in case we need to revert
> # Ask Sally: If we added a FK, what would break?
> # ACTION: why is this happening? 
> ```

@yoomlam Yes, the missing EPEs were due to a bug discovered last year that has since been resolved and the missing Decision Issues were due to manual edits outside of Caseflow. The data has been updated and is clean in all environments

File: spec/models/request_issue_spec.rb [hunk 1]
```diff
@@ -1349,7 +1349,6 @@
 
     context "when duplicate request issue is a correction" do
       let(:correction_type) { "control" }
-      let(:contested_decision_issue_id) { 12 }
```
Comment (wdrougas-tista) [inline spec/models/request_issue_spec.rb:1352]: This ID was not being used in this block

**Review by yoomlam — APPROVED** [review]
🎉 Nice work!

**Pull request merged**